tests/versioning_test.go

```diff
@@ -1591,19 +1596,28 @@ func (s *versioningIntegSuite) TestDisableUserData_WorkflowGetsStuck() {
 	s.Require().Equal(int32(1), runs.Load())
 }
 
-func (s *versioningIntegSuite) TestDisableUserData_QueryTimesOut() {
+func (s *versioningIntegSuite) TestDisableUserData_QueryFails() {
+	// force one partition so that we can unload the task queue
+	dc := s.testCluster.host.dcClient
+	dc.OverrideValue(dynamicconfig.MatchingNumTaskqueueReadPartitions, 1)
+	dc.OverrideValue(dynamicconfig.MatchingNumTaskqueueWritePartitions, 1)
+	defer dc.RemoveOverride(dynamicconfig.MatchingNumTaskqueueReadPartitions)
+	defer dc.RemoveOverride(dynamicconfig.MatchingNumTaskqueueWritePartitions)
+
 	tq := s.T().Name()
 	v1 := s.prefixed("v1")
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 	defer cancel()
+
 	s.addNewDefaultBuildId(ctx, tq, v1)
 
 	var runs atomic.Int32
 	wf := func(ctx workflow.Context) error {
-		return workflow.SetQueryHandler(ctx, "query", func() (string, error) {
```

Comment:
Why remove this?

Reply:
It confused me when I was messing with the test.. I added code after that and expected it to run. I think it's clearer to separate `SetQueryHandler`  and have the workflow do an explicit return